File: Practice/React-Native/FullNativeModules/ios/Counter.m
//
//  Counter.m
//  FullNativeModules
//
//  Created by Vladislav Onatskyi on 10/4/22.
//

#import <Foundation/Foundation.h>

  
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"


@interface RCT_EXTERN_MODULE(Counter, RCTEventEmitter)

RCT_EXTERN_METHOD(increment: (RCTResponseSenderBlock)callback)


@end

File: Practica/8-React-Native/FullNativeModules/ios/Counter.swift
//
//  Counter.swift
//  FullNativeModules
//
//  Created by Vladislav Onatskyi on 10/4/22.
//

import Foundation

@objc(Counter) 
class Counter: RCTEventEmitter {
  private var  count = 0;
  
  @objc
  func increment(_ callback: RCTResponseSenderBlock) {
    count += 1
    print("count is test")
    callback([count])
    sendEvent(withName: "onIncrement", body: ["count": count])
  }
  
  override func supportedEvents() -> [String]! {
    return ["onIncrement"]
  }
  
}
